File: src/main/java/calebzhou/rdi/core/client/util/AwtImageUtils.java
package calebzhou.rdi.core.client.util;

import calebzhou.rdi.core.client.RdiCore;

import javax.swing.*;
import java.awt.*;
import java.net.URL;

public class AwtImageUtils {
    public static Image createImage(String path, String description) {
        URL imageURL = RdiCore.class.getResource(path);

        if (imageURL == null) {
            System.err.println("Resource not found: " + path);
            return null;
        } else {
            return (new ImageIcon(imageURL, description)).getImage();
        }
    }
}

File: src/main/java/calebzhou/rdi/core/client/util/AwtImageUtils.kt
package calebzhou.rdi.core.client.util

import calebzhou.rdi.core.client.RdiCore
import java.awt.Image
import javax.swing.ImageIcon

object AwtImageUtils {
    @JvmStatic
	fun createImage(path: String, description: String?): Image? {
        val imageURL = RdiCore::class.java.getResource(path)
        return if (imageURL == null) {
            System.err.println("Resource not found: $path")
            null
        } else {
            ImageIcon(imageURL, description).image
        }
    }
}
